--- tvips/CMakeLists.txt.orig
+++ tvips/CMakeLists.txt
@@ -337,14 +337,14 @@ add_custom_target(pdf
 enable_testing()
 add_test(tiff2smv ${CMAKE_COMMAND}
     "-DCMD=${PROJECT_BINARY_DIR}/tiff2smv"
-    "-DMD5_000=9f2a26ce7a2f7827a4d6e6bb6d4000e2"
+    "-DMD5_001=7ff4a6be071ccd7844207d58f0838926"
     "-DFILE=${PROJECT_SOURCE_DIR}/test/movie23_000.tif"
     -P "${PROJECT_SOURCE_DIR}/test/test_md5.cmake"
   WORKING_DIRECTORY "${PROJECT_BINARY_DIR}")
 add_test(tvips2smv ${CMAKE_COMMAND}
     "-DCMD=${PROJECT_BINARY_DIR}/tvips2smv"
-    "-DMD5_000=9f2a26ce7a2f7827a4d6e6bb6d4000e2"
-    "-DMD5_001=6490c3aeb80a94c57d72c1ebe7f79e1a"
+    "-DMD5_000=29a3d5c42f0a2a05312a587934f1ab76"
+    "-DMD5_001=5b8e05d443eea3ebc0a6c62965137d30"
     "-DFILE=${PROJECT_SOURCE_DIR}/test/movie23_000.tvips"
     -P "${PROJECT_SOURCE_DIR}/test/test_md5.cmake"
   WORKING_DIRECTORY "${PROJECT_BINARY_DIR}")
--- tvips/test/test_md5.cmake.orig
+++ tvips/test/test_md5.cmake
@@ -12,7 +12,7 @@
 # savings.  "America/New_York" works on Unix-like systems, but not on
 # Windows.
 execute_process(
-  COMMAND "${CMD}" -a -o "t_###.img" -z "EST5EDT" "${FILE}"
+  COMMAND "${CMD}" -o "t_###.img" -z "EST5EDT" "${FILE}"
   RESULT_VARIABLE result)
 if(NOT result EQUAL "0")
   message(SEND_ERROR "Runtime error")
